pkg/xds/generator/tracing_proxy_generator.go

```diff
@@ -81,16 +83,38 @@ func (t TracingProxyGenerator) datadogCluster(backend *mesh_proto.TracingBackend
 		return nil, errors.Wrap(err, "could not convert backend")
 	}
 
-	if cfg.Port > 0xFFFF || cfg.Port < 1 {
-		return nil, errors.Errorf("invalid Datadog port number %d. Must be in range 1-65535", cfg.Port)
-	}
+	var cluster envoy.NamedResource
+	var err error
 
 	clusterName := names.GetTracingClusterName(backend.Name)
-	cluster, err := clusters.NewClusterBuilder(apiVersion).
-		Configure(clusters.DNSCluster(clusterName, cfg.Address, cfg.Port)).
-		Build()
-	if err != nil {
-		return nil, err
+
+	if strings.HasPrefix(cfg.Address, "unix:") {
```

Comment:
I think the same comment from @jpeach is applied here as well? https://github.com/kumahq/kuma/pull/2345/files#r667558085